gradle-server/src/main/java/com/github/badsyntax/gradle/GradleProjectConnector.java

```diff
@@ -55,4 +67,20 @@ public static GradleProjectConnectionType getConnectionType() {
 	public static String getLocalInstallation() {
 		return GradleProjectConnector.localInstallation;
 	}
+
+	public static File getSystemGradleHome() {
+		Map<String, String> env = System.getenv();
+		Properties sysProperties = System.getProperties();
+		String gradleHome = env.get(GRADLE_HOME);
+		if (gradleHome == null || !new File(gradleHome).isDirectory()) {
+			gradleHome = sysProperties.getProperty(GRADLE_HOME);
+		}
+		if (gradleHome != null) {
```

Comment:
here you check whether GRADLE_HOME is a directory. if it's set but is a file, will you notify user about it.

In other word, are you going to distinguish a) value not set, b) set to a file instead of a directory ?

Reply:
Currently no. Here I just use the similar logics as https://github.com/eclipse/eclipse.jdt.ls/blob/65b4728c2eaaeb286784bfdc42516fbb2818a5fc/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java#L267

Come back to this topic, if this env is point to a file, not directory, does it make sense to send a warning notification to the user about it?

Reply:
it depends on what's written in the notification. If it's saying "you don't have GRADLE_HOME` set" when user does have the env but with wrong value, then it might be confusing.

BTW, I guess it's more likely that user specify the `bin` folder just like JAVA_HOME? And my point is, just make sure the error message presenting to users is precise.

Reply:
> BTW, I guess it's more likely that user specify the bin folder just like JAVA_HOME?

Maybe, so another suggestion is we can just always use this configuration when set, and remove the check about file or directory. If the configuration is pointing to an error place, the user will get an error message and in the output channel.